src/app/api/cronjob/route.ts

```diff
@@ -0,0 +1,177 @@
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+import { addDays, eachDayOfInterval, format, isWithinInterval } from "date-fns";
+import { and, eq, gte, lte } from "drizzle-orm";
+
+import { db } from "@/lib/db";
+import {
+  gameSessions,
+  gameSessionSchedules,
+  semesters,
+  weekdayEnum,
+} from "@/lib/db/schema";
+import { getWeekday } from "@/lib/utils";
+import { insertGameSessionSchema } from "@/lib/validators";
+import { Weekday } from "@/types/types";
+
+export async function POST(req: NextRequest) {
+  try {
+    // get current day, correct to NZ timezone and reset to 12am
+    let now: Date = new Date();
+    now = new Date(now.getTime() - now.getTimezoneOffset() * 60000);
+    now = new Date(now.getTime() - (now.getTime() % (3600 * 24 * 1000)));
+
+    // Testing date
+    //now = new Date(2024, 1, 21);
+    //now = new Date(2024, 1, 22);
+    //now = new Date(2024, 2, 28);
+    //now = new Date(2024, 2, 29);
+    //now = new Date(2024, 3, 7);
+    //now = new Date(2024, 3, 8);
+    //now = new Date(2024,5,24);
+    //now = new Date(2024,5,25);
+    //now = new Date(now.getTime() - now.getTimezoneOffset() * 60000);
+    //console.log(now);
+
+    const currentDateUTC = new Date(
+      now.getTime() + now.getTimezoneOffset() * 60000,
+    );
+
+    // Find current semester
+    const activeSemester = await db.query.semesters.findFirst({
+      where: and(
+        lte(
+          semesters.startDate,
+          format(addDays(currentDateUTC, 7), "yyyy-MM-dd"),
+        ),
```

Comment:
something to keep in mind when doing date comparisons is that startDate is a NZ date, but when used in a `lte` comparison it is in a UTC context